content/02-types.md

```diff
@@ -377,53 +377,62 @@ As with other inheritance relationships, child classes can be assigned to a type
 
 ```haxe
 abstract class Base {
-	public abstract function say():String;
+  public abstract function say():String;
 
-	public function new() {}
+  public function new() {}
 }
 
 class Derived extends Base {
-	public function say():String {
-		return "Hello";
-	}
+  public function say():String {
+    return "Hello";
+  }
 }
 
 class Main {
-	public static function main() {
-		var instance:Base = new Derived();
-	}
+  public static function main() {
+    var instance:Base = new Derived();
+  }
 }
-
 ```
 
 
 A class can be declared abstract even if it has no abstract methods. Even in this case it cannot be instantiated, but non-abstract child classes can be.
 
 ```haxe
 abstract class Spaceship {
-	public function whatAmI():Void {
-		trace("Spaceship");
-	}
+  public function whatAmI():Void {
+    trace("Spaceship");
+  }
 
-	public function new() {}
+  public function new() {}
 }
 
 class Rocket extends Spaceship {}
 
 class Main {
-	public static function main() {
-		// var spaceship = new Spaceship();  // Fail
-		var rocket = new Rocket(); // Succeed
-	}
+  public static function main() {
+    // var spaceship = new Spaceship();  // Fail
+    var rocket = new Rocket(); // Succeed
+  }
 }
-
 ```
 
-> ##### Trivia: Constructor
->
-> Even though they can't be instantiated, abstract classes can still have a constuctor. 
+Even though they can't be instantiated, abstract classes can still have a constructor that the child class can call with `super()`. Definition of this constructor is optional.
```

Comment:
> Definition of this constructor is optional.

Hmm, what exactly do you mean here?

Reply:
That's meant to say that the abstract class doesn't _need_ the constructor. 
Perhaps the following code should be the example:
`abstract class Parent {
}

class Child extends Parent {
	public function new() {
    trace("Child created!");
  }
}
`

Reply:
Right, I suppose that's already implied by the "can" in the previous sentence. I think this is actually clearer without the additional line because it doesn't really clarify anything and might just confuse people (like me just now, haha).

Reply:
OK, I'll drop that last sentence